### PR TITLE
shell script suggestions

### PR DESCRIPTION
**Pull request by andkaha**
Some shell scripting suggestions from me. Nothing really serious.

Mostly about quoting, trailing `;` on commands, formatting, testing for empty strings etc.

No functional changes as far as I know.

### CODE REVIEW

**Review by blankdots — APPROVED** [review]

File: extras/travis_push_docker_hub.sh [hunk 1]
```diff
@@ -1,41 +1,49 @@
...
+    retag_images latest "$TRAVIS_TAG" true
 fi
 
-if [[ "$TRAVIS_PULL_REQUEST" != "false" && "$TRAVIS_PULL_REQUEST" != "" && "$TRAVIS_BUILD_STAGE_NAME" == "Integration tests" ]]; then
-    retag_images "PR${TRAVIS_PULL_REQUEST}" "dev" false
+if  [ -n "$TRAVIS_PULL_REQUEST" ] &&
+    [ "$TRAVIS_PULL_REQUEST" != "false" ] &&
+    [ "$TRAVIS_BUILD_STAGE_NAME" = "Integration tests" ]
+then
+    retag_images "PR$TRAVIS_PULL_REQUEST" dev false
```
Comment (blankdots) [inline extras/travis_push_docker_hub.sh:48]: This is pushing image with tag `dev` and it should not:
* https://travis-ci.org/NBISweden/LocalEGA/jobs/431543758#L522
* https://travis-ci.org/NBISweden/LocalEGA/jobs/431543758#L662

I think it is related to line 12 in this file, the way `true` or `false` is evaluated.

**Review by blankdots — CHANGES_REQUESTED** [review]
Please change so that the `dev` tag is not submitted during integration tests.

File: extras/travis_push_docker_hub.sh [hunk 2]
```diff
@@ -1,41 +1,49 @@
...
     to=$3
     push=$4
-    docker pull "${base}:${from}"
-    docker tag "${base}:${from}" "${base}:${to}"
-    if ${push}; then
-      echo "Pushing LocalEGA image: ${base}:${to}"
-      docker push "${base}:${to}"
+    docker pull "$base:$from"
+    docker tag "$base:$from" "$base:$to"
+    if [ -n "$push" ]; then
```
Comment (blankdots) [inline extras/travis_push_docker_hub.sh]: I think this line should be one of the below:
* `if [ "$push" = true ]; then`
* `if $push ; then`

Comment (andkaha) [reply extras/travis_push_docker_hub.sh]: Yes, I was confused and thought you were testing for a non-empty string, not evaluating input as commands. I've pushed a commit that does the first of you two suggestions.

**Review by blankdots — APPROVED** [review]

**Review by viklund — APPROVED** [review]
Nice

**Pull request merged**